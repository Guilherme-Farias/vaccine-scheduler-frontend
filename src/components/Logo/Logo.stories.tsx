import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import Logo from '.';

export default {
  title: 'Logo',
  component: Logo,
  parameters: {
    backgrounds: {
      default: 'minha-vacina-minor',
    },
  },
} as ComponentMeta<typeof Logo>;

const Template: ComponentStory<typeof Logo> = args => <Logo {...args} />;

export const Default = Template.bind({});
Default.args = {};
