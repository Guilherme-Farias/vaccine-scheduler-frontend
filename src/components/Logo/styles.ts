import styled, { css } from 'styled-components/macro';

export const LogoWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    color: ${theme.colors.white};
    text-decoration: none;
    font-size: ${theme.font.sizes['3xl']};
    @media ${theme.queries.phoneAndSmaller} {
      font-size: ${theme.font.sizes['2xl']};
    }
  `}
`;

export const IconWrapper = styled.div`
  ${({ theme }) => css`
    margin-right: ${theme.spacings['2xs']};
    > svg {
      font-size: inherit;
    }
  `}
`;

export const Text = styled.h1`
  ${({ theme }) => css`
    font-size: inherit;
    font-weight: ${theme.font.regular};
  `}
`;
