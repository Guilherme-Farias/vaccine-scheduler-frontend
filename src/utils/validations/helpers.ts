import Joi from 'joi';
import { IParamErrors } from 'types/global';

export function getFieldErrors(objError: Joi.ValidationResult) {
  const errors: IParamErrors = {};

  if (objError.error) {
    objError.error.details.forEach(err => {
      errors[err.path.join('.')] = err.message;
    });
  }

  return errors;
}
