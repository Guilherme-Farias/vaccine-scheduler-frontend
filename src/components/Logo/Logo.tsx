import React from 'react';

import { FaSyringe as SyringeIcon } from 'react-icons/fa';
import * as S from './styles';

const Logo: React.FC = () => (
  <S.LogoWrapper>
    <S.IconWrapper>
      <SyringeIcon />
    </S.IconWrapper>
    <S.Text>
      Minha <strong>Vacina</strong>
    </S.Text>
  </S.LogoWrapper>
);

export default Logo;
