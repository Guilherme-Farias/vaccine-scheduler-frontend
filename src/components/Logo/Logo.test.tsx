import { render, screen } from 'utils/tests/helpers';

import Logo from '.';

describe('<Logo />', () => {
  it('should render the Logo', () => {
    const { container } = render(<Logo />);
    expect(
      screen.getByRole('heading', { name: /minha vacina/i }),
    ).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
