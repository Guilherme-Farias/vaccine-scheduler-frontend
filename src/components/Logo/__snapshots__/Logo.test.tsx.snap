// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render the Logo 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.875rem;
}

.c1 {
  margin-right: 0.25rem;
}

.c1 > svg {
  font-size: inherit;
}

.c2 {
  font-size: inherit;
  font-weight: 400;
}

@media (max-width:48rem) {
  .c0 {
    font-size: 1.5rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M201.5 174.8l55.7 55.8c3.1 3.1 3.1 8.2 0 11.3l-11.3 11.3c-3.1 3.1-8.2 3.1-11.3 0l-55.7-55.8-45.3 45.3 55.8 55.8c3.1 3.1 3.1 8.2 0 11.3l-11.3 11.3c-3.1 3.1-8.2 3.1-11.3 0L111 265.2l-26.4 26.4c-17.3 17.3-25.6 41.1-23 65.4l7.1 63.6L2.3 487c-3.1 3.1-3.1 8.2 0 11.3l11.3 11.3c3.1 3.1 8.2 3.1 11.3 0l66.3-66.3 63.6 7.1c23.9 2.6 47.9-5.4 65.4-23l181.9-181.9-135.7-135.7-64.9 65zm308.2-93.3L430.5 2.3c-3.1-3.1-8.2-3.1-11.3 0l-11.3 11.3c-3.1 3.1-3.1 8.2 0 11.3l28.3 28.3-45.3 45.3-56.6-56.6-17-17c-3.1-3.1-8.2-3.1-11.3 0l-33.9 33.9c-3.1 3.1-3.1 8.2 0 11.3l17 17L424.8 223l17 17c3.1 3.1 8.2 3.1 11.3 0l33.9-34c3.1-3.1 3.1-8.2 0-11.3l-73.5-73.5 45.3-45.3 28.3 28.3c3.1 3.1 8.2 3.1 11.3 0l11.3-11.3c3.1-3.2 3.1-8.2 0-11.4z"
      />
    </svg>
  </div>
  <h1
    class="c2"
  >
    Minha 
    <strong>
      Vacina
    </strong>
  </h1>
</div>
`;
